src/content_script/mastodonInject.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * @typedef {Object} VersionNumber
+ * @property {string} major
+ * @property {string} minor
+ * @property {string} patch
+ */
+
+/**
+ * parses a versionNumber string
+ * @returns {VersionNumber}
+ */
+function parseVersionNumber() {
+  const versionElement = document.querySelector(`.link-footer`).textContent
+  const versionNumber = versionElement.match(/v?(?<major>\d+)\.(?<minor>\d+)\.(?<patch>\d+)$/)
+  return versionNumber.groups
+}
+
+
+/**
+ * Replacement onClick handler for Follow button
+ * @param {Event} event 
+ */
+function onClickFollow (event) {
+  event.stopPropagation()
+  event.preventDefault()
+  const username = window.location.pathname.split(`/`).slice(-1)[0]
+  // activate AutoRemoteFollow
+  window.open(`/users/${username}/remote_follow`, `_blank`)
+}
+
+/**
+ * wait for element to appear
+ * @param {string} selector
+ * @param {number} timeout
+ */
+function waitForElement (selector, timeout) {
+  return new Promise((resolve, reject) => {
+    function getElement() {
+      return document.querySelector(selector)
+    }
+
+    const element = getElement()
+    if(element){
+      return resolve(element)
+    }
+
+    const observer = new MutationObserver(() => {
+      const element = getElement()
+      if(element){
+        resolve(element)
+        observer.disconnect()
+      }
+    })
+
+    observer.observe(document.body, {
+      childList: true,
+      subtree: true
+    })
+
+    window.setTimeout(() => {
+      reject()
+    }, timeout)
+  }) 
+}
+
+/**
+ * Inject replacement onClick handler for Follow button
+ */
+async function injectFollowButton () {
+  try {
+    const followButton = await waitForElement(`.account__header__tabs__buttons button.button`, 20000)
+    followButton.addEventListener(`click`, onClickFollow)
```

Comment:
BTW, could not we also just modify the HTML, i.e. instead of an event listener and JS, add a HTML -> `href` and `target` as usual?

Did you test that and why did you decide for that solution/JS-way here?

IMHO, the HTML version could be safer/I'd choose that way intuitively (and also understood your comment like that before).

Reply:
That would equal the way is was before v4 then:

![image](https://user-images.githubusercontent.com/11966684/202274952-91b89b89-e56a-428e-b8d6-dec49a1f9c39.png)

Here an example: https://archive.ph/gcSGv

(original [archived from this link](https://chaos.social/@rugk/103810871178820703))

```html
<a href="https://archive.ph/o/gcSGv/https://chaos.social/users/rugk/remote_follow" target="_blank" style="background-position:initial;background-repeat:initial;-webkit-background-clip:border-box;color:rgb(255, 255, 255);font-family:mastodon-font-sans-serif, sans-serif;font-size:14px;font-stretch:100%;font-style:normal;font-variant-caps:normal;font-variant-east-asian:normal;font-variant-ligatures:normal;font-variant-numeric:normal;font-weight:500;align-items:center;background-attachment:scroll;background-clip:border-box;background-color:rgb(43, 144, 217);background-image:none;background-origin:padding-box;background-size:auto;border-bottom-color:rgb(255, 255, 255);border-bottom-left-radius:4px;border-bottom-right-radius:4px;border-bottom-style:none;border-bottom-width:0px;border-image-outset:0;border-image-repeat:stretch;border-image-slice:100%;border-image-source:none;border-image-width:1;border-left-color:rgb(255, 255, 255);border-left-style:none;border-left-width:0px;border-right-color:rgb(255, 255, 255);border-right-style:none;border-right-width:0px;border-top-color:rgb(255, 255, 255);border-top-left-radius:4px;border-top-right-radius:4px;border-top-style:none;border-top-width:0px;box-sizing:border-box;cursor:pointer;display:flex;flex-basis:auto;flex-grow:0;flex-shrink:1;height:auto;hyphens:auto;justify-content:center;letter-spacing:normal;line-height:1.2;margin-bottom:15px;margin-left:0px;margin-right:0px;margin-top:0px;min-height:36px;min-width:88px;overflow-wrap:break-word;overflow-x:hidden;overflow-y:hidden;padding-bottom:0px;padding-left:15px;padding-right:15px;padding-top:0px;position:relative;text-align:center;text-decoration-color:rgb(255, 255, 255);text-decoration-line:none;text-decoration-style:solid;text-overflow:ellipsis;text-transform:none;vertical-align:baseline;white-space:normal;width:auto;text-decoration-thickness:auto;"><svg viewBox="0 0 216.4144 232.00976" style="min-height:21.4375px;min-width:20px;fill:rgb(255, 255, 255);height:21.4375px;margin-right:5px;overflow-x:hidden;overflow-y:hidden;vertical-align:middle;width:20px;"><use xlink:href="#mastodon-svg-logo" style="transform-origin:0px 0px;"></use></svg>Follow</a>
```

(okay way too much, but u get the point)

Reply:
The issue I had with adding a `href` attribute is that the follow button is actually `<button>` now, which doesn't support the `href` attribute. We could delete the entire `<button>` and replace it with an `<a href"...">` but it does seem more complicated and styling could be affected.

I'm happy to try that instead if you think it's safer.

Reply:
Okay no, given that explanation, a JS handler seems better.

Also accessibility-wise, `button` is the better HTML markup anyway, so I guess this is also why they switched.
So no, let's use it as it is.